Don't allow to run cli tools as root user

bjoernricks opened the pull request:
Add function that raises a RuntimeError if one of our tools is run a
root user. Running the tool as root is never required and is always a
bad habit.

**Checklist**:

- [x] [CHANGELOG](https://github.com/greenbone/gvm-tools/blob/master/CHANGELOG.md) Entry

wiegandm commented:
I am a bit conflicted regarding categorically denying this.

I fully agree that it is completely unnecessary to run tools like `gvm-cli` as
`root` and I think there should be at least a strongly worded warning, nudging
users towards the 'right thing'.

However, simply aborting without any opportunity to override seems a bit harsh
to me.

Also, I am not quite sure how `os.getuid()` would react on non-Unix platforms.
Can you shed some light there?

bjoernricks commented:
> Also, I am not quite sure how os.getuid() would react on non-Unix platforms.
> Can you shed some light there?

Personally I thought getuid would return None in that case but i guess the function is not available at all. Therefore I changed the code to check for the function first.

> However, simply aborting without any opportunity to override seems a bit harsh
> to me.

User **MUST NOT** run any of our python code as root user. I really don't want them to be able to do that. It's stupid and insecure. It's a bad habit from *i don't know how to use permissions correctly so i use sudo*...

I've changed the check from using getuid to geteuid too.
